packages/aws-cdk-lib/aws-codepipeline/lib/pipeline.ts

```diff
@@ -373,6 +481,8 @@ export class Pipeline extends PipelineBase {
   private readonly enableKeyRotation?: boolean;
   private readonly reuseCrossRegionSupportStacks: boolean;
   private readonly codePipeline: CfnPipeline;
+  private readonly pipelineType?: PipelineType;
```

Comment:
To prevent changes from occurring with the existing version of the template, this parameter is made optional rather than overwritten by V1 when unspecified in props.

Reply:
We can always set the the internal prop, just don't set it on the CfnPipeline.

Reply:
Let's also throw a dismissible warning if pipeline type is not set.

Reply:
@mrgrain 

I changed them.

> We can always set the the internal prop, just don't set it on the CfnPipeline.

https://github.com/aws/aws-cdk/pull/28538/commits/3e799832c7f6d1f7d69613570afa17dca35ad727

> Let's also throw a dismissible warning if pipeline type is not set.

https://github.com/aws/aws-cdk/pull/28538/commits/529786ad335c5514937d9a0e43ff3b645bf10480